task/types.go

```diff
@@ -4,8 +4,6 @@
 
 package task
 
-import "encoding/json"
-
 type Task struct {
 	// ID provides a unique task identifier.
 	ID string `json:"id"`
```

Comment:
we can remove this id

we have task id defined at all 3 levels, very easy to make mistakes. At DelegateTask level, at RunnerRequest, and at Task

Reply:
The `ID` defined here is being used in `router.go::Handle` function, for mapping values of decrypted secrets. I think we could change it to not use the `ID` and then get this field removed, but I would prefer doing it in a separate PR, what do you think?